### PR TITLE
WIP of making a smaller petstore

### PR DESCRIPTION
**Pull request by posix4e**
https://github.com/posix4e/petstore-jsoncdc Is my example petstore. It seems much smaller :)

Plenty still to do. I need to make a fancy writer function. I'm convinced a little macro magic can slay the VAR ARG FFI problem. 

I've hardcoded to 95 because our cfg 94/95 switch doesn't get passed to the module. I think this is fine anyway?

Tests need to pass. I did try this locally and it seems to be installing and working.

### CODE REVIEW

**Comment by posix4e** [comment]
We probably want to break the pgrustxn out into different versions. We could leave this repo as the pet store. But i wanted to get your feedback

File: pgrustxn/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "pgrustxn"  #til i get a better name
+license = "Apache-2.0"
+version = "0.0.7"
+authors = ["Alex Newman <posi@planet.com>"] #TODO get jason to add his name :)
+description   = "Lightweight non-blocking IO"
```
Comment (solidsnack) [inline pgrustxn/Cargo.toml:6]: "Lightweight non-block IO" ? ;)

File: pgrustxn/src/decoder.rs [hunk 2]
```diff
@@ -1,95 +1,46 @@
...
-    use pg::Enum_ReorderBufferChangeType::*;
     let tuple_desc = (*relation).rd_att;
     let tuples = (*change).data.tp();
     let tuple_new = (*tuples).newtuple;
     let tuple_old = (*tuples).oldtuple;
     let token = match (*change).action {
-        REORDER_BUFFER_CHANGE_INSERT => "insert",
-        REORDER_BUFFER_CHANGE_UPDATE => "update",
-        REORDER_BUFFER_CHANGE_DELETE => "delete",
+        pg::Enum_ReorderBufferChangeType::REORDER_BUFFER_CHANGE_INSERT => "insert",
```
Comment (solidsnack) [inline pgrustxn/src/decoder.rs:41]: This will fail the `rustfmt` pass.

File: pgrustxn/src/texttools.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
+#[cfg(pg94)]
+pub use pg94 as pg;
+
+//#[cfg(pg95)]
+pub use pg95 as pg;
+
+use std::ffi::CString;
+use std::mem::size_of;
+
+pub fn write_text1(ctx: *mut pg::Struct_LogicalDecodingContext,
```
Comment (solidsnack) [inline pgrustxn/src/texttools.rs:10]: This is a good direction to go. Is `t` a text object? Like a `StringInfo` or something like that?

File: util/travis [hunk 4]
```diff
@@ -38,7 +38,7 @@ function linux {
   HBA_CONF="$PG_DIR"/pg_hba.conf
   CONF="$PG_DIR"/postgresql.conf
 
-  sudo -E env PATH="$PATH" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" make install
+  sudo -E env PATH="$PATH" LD_LIBRARY_PATH="$LD_LIBRARY_PATH" (cd jsoncdc; make install)
```
Comment (solidsnack) [inline util/travis:41]: I am starting to think this is a big enough experiment that a repo-level separation might be good. The `PostgresRSPetStore` is a challenging project and we might have to try a bunch of different ways to get a story for modularity and reuse that we like.

File: pgrustxn/src/texttools.rs (hunk 3 above)
Comment (posix4e) [inline pgrustxn/src/texttools.rs:10]: right

**Pull request closed without merging**